yt_dlp/extractor/nhk.py

```diff
@@ -28,6 +28,16 @@ def _call_api(self, m_id, lang, is_video, is_episode, is_clip):
                 m_id, lang, '/all' if is_video else ''),
             m_id, query={'apikey': 'EJfK8jdS57GqlupFgAfAAwr573q01y6k'})['data']['episodes'] or []
 
+    def _get_vod_api(self, vod_id):
+        movie_player_js = self._download_webpage(
+            'https://movie-a.nhk.or.jp/world/player/js/movie-player.js', vod_id,
+            note='Downloading stream API information')
+        api_url = self._search_regex(
+            r'prod:[^;]+apiUrl:\s*[\'"]([^\'"]+)[\'"]', movie_player_js, vod_id, 'stream API url')
+        api_token = self._search_regex(
+            r'prod:[^;]+token:\s*[\'"]([^\'"]+)[\'"]', movie_player_js, vod_id, 'stream API token')
+        return api_url, api_token
```

Comment:
Since this si requesting "static" parameters, this should imo not be tied to `vod_id` and instead be using a video_id of `None`. Also perhaps it should be cached so we wouldnt download it more than once? How long before the api url and api token expire?

Reply:
> Since this si requesting "static" parameters, this should imo not be tied to `vod_id` and instead be using a video_id of `None`.

yes, will do :+1: 

> How long before the api url and api token expire?

don't know, not entirely sure they do

> Also perhaps it should be cached so we wouldnt download it more than once?

i think yes but actually no, see pr description
fwiw ive done this same thing in the `tbsjp` extractor and no complaints were raised then

Reply:
Yes, I seem to have misjudged what you were trying to do and you were right in your gut feeling.
Please ignore the comment in this context, especially since expiration time →∞ (most likely)

Reply:
Sorry, maybe my last comment was written in a somewhat ambiguous way, I meant we ***should*** do caching.

Reply:
oh, right

Reply:
901f8c55a8a979ffce7bf6ac46e304c2261418f7 hows this?
(or 8c9dee3f6093296c9ff5e85b7040bfc0169d0ab1 rather)

Reply:
@bashonly and I revised the caching strategy a bit